src/grt/src/fastroute/include/FastRoute.h

```diff
@@ -120,28 +125,344 @@ class FastRouteCore
+  bool netCongestion(int netID);
+
+  // route functions
+  // old functions for segment list data structure
+  void routeSegL(Segment* seg);
+  void routeLAll(bool firstTime);
+  // new functions for tree data structure
+  void newrouteL(int netID, RouteType ripuptype, bool viaGuided);
+  void newrouteZ(int netID, int threshold);
+  void newrouteZ_edge(int netID, int edgeID);
+  void newrouteLAll(bool firstTime, bool viaGuided);
+  void newrouteZAll(int threshold);
+  void routeMonotonicAll(int threshold);
+  void routeMonotonic(int netID, int edgeID, int threshold);
+  void routeLVAll(int threshold, int expand, float logis_cof);
+  void spiralRouteAll();
+  void newrouteLInMaze(int netID);
+  void estimateOneSeg(Segment* seg);
+  void routeSegV(Segment* seg);
+  void routeSegH(Segment* seg);
+  void routeSegLFirstTime(Segment* seg);
+  void spiralRoute(int netID, int edgeID);
+  void routeLVEnew(int netID, int edgeID, int threshold, int enlarge);
+
+  // ripup functions
+  void ripupSegL(Segment* seg);
+  void ripupSegZ(Segment* seg);
+  void newRipup(TreeEdge* treeedge,
+                       TreeNode* treenodes,
+                       int x1,
+                       int y1,
+                       int x2,
+                       int y2,
+                       int netID);
+  bool newRipupCheck(TreeEdge* treeedge,
+                            int x1,
+                            int y1,
+                            int x2,
+                            int y2,
+                            int ripup_threshold,
+                            int netID,
+                            int edgeID);
+
+  bool newRipupType2(TreeEdge* treeedge,
+                            TreeNode* treenodes,
+                            int x1,
+                            int y1,
+                            int x2,
+                            int y2,
+                            int deg,
+                            int netID);
+  bool newRipup3DType3(int netID, int edgeID);
+  void newRipupNet(int netID);
+
+  // utility functions
+  void printEdge(int netID, int edgeID);
+  void ConvertToFull3DType2();
+  void fillVIA();
+  int threeDVIA();
+  void assignEdge(int netID, int edgeID, bool processDIR);
+  void recoverEdge(int netID, int edgeID);
+  void newLayerAssignmentV4();
+  void netpinOrderInc();
+  void checkRoute3D();
+  void StNetOrder();
+  bool checkRoute2DTree(int netID);
+  void checkUsage();
+  void netedgeOrderDec(int netID);
+  void printTree2D(int netID);
+  void printEdge2D(int netID, int edgeID);
+  void printEdge3D(int netID, int edgeID);
+  void printTree3D(int netID);
+  void check2DEdgesUsage();
+  void newLA();
+  void copyBR(void);
+  void copyRS(void);
+  void freeRR(void);
+  Tree fluteToTree(stt::Tree fluteTree);
+  stt::Tree treeToFlute(Tree tree);
+  int edgeShift(Tree* t, int net);
+  int edgeShiftNew(Tree* t, int net);
+
+  static const int MAXLEN = 20000;
+  static const int BIG_INT = 1e7;     // big integer used as infinity
```

Comment:
Can you use std::numeric_limits<int>::max() here?  Initializing an int with a double is odd.

Reply:
When I changed it to std::numeric_limits::max(), my regression tests fail with an invalid layer assignment.

Reply:
yuck - that's something we should chase down but it can wait.